src/rdb.c

```diff
@@ -3562,6 +3564,11 @@ void backgroundSaveDoneHandler(int exitcode, int bysignal) {
     /* Possibly there are replicas waiting for a BGSAVE in order to be served
      * (the first stage of SYNC is a bulk transfer of dump.rdb) */
     updateReplicasWaitingBgsave((!bysignal && exitcode == 0) ? C_OK : C_ERR, type);
+    if (server.cluster_enabled && clusterIsAnySlotExporting()) {
+        /* Slot export should also be notified, in case this was a export
+         * related snapshot */
+        clusterHandleSlotExportBackgroundSaveDone((!bysignal && exitcode == 0) ? C_OK : C_ERR);
+    }
```

Comment:
the check of `server.cluster_enabled && clusterIsAnySlotExporting()` feels out of place. it'd be cleaner if we move them into `clusterHandleSlotExportBackgroundSaveDone` instead like how `updateReplicasWaitingBgsave` handles the possibility of replicas full sync case.

```suggestion
    /* Slot export should also be notified, in case this was a export
     * related snapshot */
    clusterHandleSlotExportBackgroundSaveDone((!bysignal && exitcode == 0) ? C_OK : C_ERR);
```

Reply:
Done